### ISSUE TITLE
Can't save profile page changes

### ISSUE THREAD

**Issue by ainsleyclark** [strapi/strapi#18678]
### Required System information

- Node.js version: v18.18.2
- NPM version: 9.8.1
- Strapi version: 4.15.1
- Database: Postgres 12
- Operating system: node:18-alpine
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When I try and change the interface mode to `Light Mode` it says `Warning: an error occured`

### Steps to reproduce the behavior

1. Go to 'User Profile'
2. Try to switch to 'Light Mode' from theme settings.
3. No change is observed, and the interface remains in 'Dark Mode'.
4. No error message is displayed in the UI.

### Expected behavior

I expect the interface to switch to light mode when I select the option in the user profile settings.

### Screenshots

<img width="1225" alt="image" src="https://github.com/strapi/strapi/assets/34712954/c23ddd60-1752-499c-8390-10a62799cc4c">

<img width="677" alt="image" src="https://github.com/strapi/strapi/assets/34712954/9de4e8ec-73b9-4baf-ba5f-21565a84bb52">

### Code snippets

```
[strapi] [2023-11-06 08:22:18] ValidationError: 5 errors occurred
[strapi] [2023-11-06 08:22:18]     at handleYupError (/opt/node_modules/@strapi/admin/node_modules/@strapi/utils/dist/index.js:408:9)
[strapi] [2023-11-06 08:22:18]     at /opt/node_modules/@strapi/admin/node_modules/@strapi/utils/dist/index.js:418:7
[strapi] [2023-11-06 08:22:18]     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[strapi] [2023-11-06 08:22:18]     at async Object.updateMe (/opt/node_modules/@strapi/admin/server/controllers/authenticated-user.js:18:5)
[strapi] [2023-11-06 08:22:18]     at async returnBodyMiddleware (/opt/node_modules/@strapi/strapi/dist/services/server/compose-endpoint.js:42:18)
[strapi] [2023-11-06 08:22:18]     at async policiesMiddleware (/opt/node_modules/@strapi/strapi/dist/services/server/policy.js:15:5)
[strapi] [2023-11-06 08:22:18]     at async /opt/node_modules/@strapi/strapi/dist/services/server/compose-endpoint.js:27:12
[strapi] [2023-11-06 08:22:18]     at async /opt/node_modules/@strapi/strapi/dist/middlewares/body.js:45:9
[strapi] [2023-11-06 08:22:18]     at async /opt/node_modules/@strapi/strapi/dist/middlewares/logger.js:6:5
[strapi] [2023-11-06 08:22:18]     at async /opt/node_modules/@strapi/strapi/dist/middlewares/powered-by.js:9:5
```

**Comment by Eventyret** [comment]
This version was deprivated could you please try version 4.14.6
https://github.com/strapi/strapi/releases/tag/v4.14.6

**Comment by ainsleyclark** [comment]
Downgrading to `4.14.6` has fixed the issue.

**Comment by remidej** [comment]
@derrickmehaffy I believe this is still an issue on 4.15.2, should we reopen?

**Comment by derrickmehaffy** [comment]
> @derrickmehaffy I believe this is still an issue on 4.15.2, should we reopen?

Absolutely

**Comment by BasharOsman** [comment]
same error on v4.15.4

**Comment by remidej** [comment]
@derrickmehaffy I know we're busy with several issues currently, but I feel like this one deserves more attention. It's not just the theme that can't be changed, it's everything on the profile page (except password probably)

**Comment by BasharOsman** [comment]
if you don't change the password the response will be ValidationError: 5 errors occurred
if you add the password and the re password you can't save and the update request not send

**Comment by derrickmehaffy** [comment]
After discussing with @alexandrebodin  we have a temporary policy for the change freeze so I'll add this one to it @remidej thanks for raising this and I'll jump the severity.

**Comment by BasharOsman** [comment]
@derrickmehaffy there other problem here that if you want to add translations from admin/app.js fro example add "fr" langauge then build the app nothing will change in language list just include English language  the error tested on 4.15.4
tested on 4.14.6 no problem

**Comment by remidej** [comment]
@BasharOsman that sounds unrelated, could you open a new issue? If there isn't one of course

**Comment by gu-stav** [comment]
This may have been broken in https://github.com/strapi/strapi/pull/18593

**Comment by remidej** [comment]
@gu-stav I'm currently working on a fix

**Comment by gu-stav** [comment]
@remidej @joshuaellis is working on it too, as the issue is assigned to him and the platform squad.

**Comment by remidej** [comment]
@gu-stav Josh told me to work on it yesterday 🙃 

my PR is up (#18789), do what you want with it

**Comment by joshuaellis** [comment]
I think there's just some communication confusion.

Next time, I would assign yourself so we know you're dealing with it, equally, I should have spoken to you, but I forgot, sorry both for that – I didn't get too far so let's go with your solution anyway.

Thanks for doing it regardless!

**Comment by martinjdlabs** [comment]
I still have this issue on "@strapi/strapi": "4.15.4"

**Comment by t-fritsch** [comment]
@martinjdlabs that's already been said 2 weeks ago, here : https://github.com/strapi/strapi/issues/18678#issuecomment-1808604514 ...

**Comment by alexandrebodin** [comment]
Hey the fix is ready to be release in 4.15.5. It should be release next wednesday. We were waiting to release a few other fixes at the same time

**Comment by derrickmehaffy** [comment]
Related to internal TID: 5753

When debugging an SSO issue with a customer on v4.19.0 this same issue came up when trying to change the roles where the password was required, likewise I believe for SSO users we may need to remove the password option entirely if the user was using SSO but that's a topic we can handle later.

TLDR: As of v4.19.0 users can't edit another user (change roles) without setting a password which conflicts with SSO.
